src/qibosoq/programs/base.py

```diff
@@ -127,6 +127,7 @@ def add_pulse_to_register(self, pulse: Pulse):
             elif is_drag:
                 # delta will be divided for the same quantity, we are setting it = 1
                 delta = self.soccfg["gens"][gen_ch]["samps_per_clk"] * self.soccfg["gens"][gen_ch]["f_fabric"]
+                delta = -delta / 2
```

Comment:
```suggestion
                delta = - self.soccfg["gens"][gen_ch]["samps_per_clk"] * self.soccfg["gens"][gen_ch]["f_fabric"] / 2
```

It's just the same, but you avoid giving the same name to two different things.